packages/flutter/lib/src/widgets/title.dart

```diff
@@ -28,10 +33,22 @@ class Title extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    _updateSystemChrome();
     updateTaskDescription(label: title, color: color);
     return child;
   }
 
+  /// Updates the system chrome settings based on this title's metadata.
+  void _updateSystemChrome() {
+    // TODO(tvolkert): This may result in a decent amount of unnecessary
+    // overhead in the embedder (on every build). Consider making Title
+    // a StatefulWidget that only calls into the SystemChrome service when
+    // it sees that a value has changed.
+    SystemChrome.setSystemUIOverlayStyle(brightness == Brightness.dark
+        ? mojom.SystemUiOverlayStyle.light
+        : mojom.SystemUiOverlayStyle.dark);
+  }
```

Comment:
Yeah, `updateTaskDescription` has logic to filter out redundant writes in this way.